failed compile with  -DBUILD_ZIQ=OFF

**Description of the issue**
When building with `-DBUILD_ZIQ=OFF` the compile fails, snippet of the logs below.

**Hardware (SDR/PC/OS)**
[docker image](https://github.com/rpatel3001/docker-satdump) based on debian:bookworm-20240904-slim

**Version (Eg, 1.0.0, CI Build 171)**
current master, hash 8e42faa86e04426a83f0c693a026b2c83af68d94

**Logs after the crash** (satdump.log from ~/.config/satdump or %appdata%\satdump)
```
1188.4 /src/satdump/src-cli/record.cpp: In function 'int main_record(int, char**)':
1188.4 /src/satdump/src-cli/record.cpp:186:23: error: 'class dsp::BasebandType' has no member named 'ziq_depth'
1188.4   186 |         baseband_type.ziq_depth = parameters["ziq_depth"].get<int>();
1188.4       |                       ^~~~~~~~~
1188.4 /src/satdump/src-cli/record.cpp:189:58: error: 'class dsp::BasebandType' has no member named 'ziq_depth'
1188.4   189 |         logger->info("Using ZIQ Depth %d", baseband_type.ziq_depth);
1188.4       |                                                          ^~~~~~~~~
```

**Other info (Eg, Screenshots) / Files useful for debugging (CADU, etc)**
Based on [this line](https://github.com/SatDump/SatDump/blob/8e42faa86e04426a83f0c693a026b2c83af68d94/src-core/common/dsp/io/baseband_type.h#L42), there needs to be some preprocessor gating around [these couple lines](https://github.com/SatDump/SatDump/blob/8e42faa86e04426a83f0c693a026b2c83af68d94/src-cli/record.cpp#L185). possibly similar to #807

--- a/src-cli/record.cpp
+++ b/src-cli/record.cpp
@@ -182,11 +182,13 @@ int main_record(int argc, char *argv[])
 
     dsp::BasebandType baseband_type = parameters["baseband_format"].get<std::string>();
 
+#if defined(BUILD_ZIQ) || defined(BUILD_ZIQ2)
     if (parameters.contains("ziq_depth"))
         baseband_type.ziq_depth = parameters["ziq_depth"].get<int>();
 
     if (parameters["baseband_format"].get<std::string>() == "ziq")
         logger->info("Using ZIQ Depth %d", baseband_type.ziq_depth);
+#endif
 
     file_sink->set_output_sample_type(baseband_type);
     file_sink->start();

